### DISCUSSION TITLE
Reference Theme Values - Tailwind Config

### DISCUSSION THREAD

**Discussion by zuperduper** [tailwindlabs/tailwindcss#12586] [Help] [answered]
Hello, 

I'd like to create color tokens that reference other theme colors that I've created within my tailwind config. Been really struggling to get this to work. I'm using Tailwind v3.3.6 and have followed what is noted in the docs here: https://tailwindcss.com/docs/theme#referencing-other-values

Here is a very stripped down config example where I'd like my `headline` color to reference the `primary.500` color value:

```ts
import type { Config } from 'tailwindcss'

const config: Config = {
  content: ['./src/**/*.{js,ts,jsx,tsx,mdx}'],
  theme: {
    extend: {
      colors: ({ theme }) => ({
        primary: {
          500: '#b61a81',
        },
        headline: theme('colors.primary.500'),
      }),
    },
  },
  plugins: [],
}

export default config
```

Not sure if I'm doing something wrong or if this is an actual issue? But I keep getting this error `RangeError: Maximum call stack size exceeded`. 

Appreciate any help 🙏🏼

**Comment by wongjn** [comment]
You are doing something wrong. As the error suggests, you're causing an infinite loop. This is because `theme()` tries to resolve the `colors` configuration, but to do that, Tailwind  needs to call your `colors` configuration function. However, when Tailwind does that, there is a `theme()` call inside the configuration function, so  `theme()` tries to resolve the `colors` configuration, but to do that, Tailwind  needs to call your `colors` configuration function. However, when Tailwind does that, there is a `theme()` call inside the configuration function, so…

**Reply by alex-krasikau** [reply, marked as answer]
Hey @zuperduper!

You can't use the same object as both a value and a reference. To fix this, extract the base palette into a separate object and use it to define the color scheme. Here's an example:

```
const colors = {
  primary: {
     500: '#b61a81',
  },
};

const config: Config = {
  content: ['./src/**/*.{js,ts,jsx,tsx,mdx}'],
  theme: {
    extend: {
      colors: ({ theme }) => ({
        ...colors,
        headline: colors.primary.500,
      }),
    },
  },
  plugins: [],
}

```

The `theme` function will work for any other configurations you have.

**Reply by zuperduper** [reply]
Ah I see - thanks for the help guys, appreciate you! :)
